alphalens/tests/test_performance.py

```diff
@@ -835,70 +772,75 @@ def test_factor_rank_autocorrelation(self,
-                             [0.075, 0.0], [0.075, 0.0], [0.075, 0.0],
-                             [0.075, 0.0], [0.075, 0.0], [0.075, 0.0],
-                             [0.075, 0.0], [0.075, 0.0], [0.075, 0.0],
-                             [0.075, 0.0]]),
-                           (6, 6, True, True,
-                            [[0., 0.], [0., 0.], [0., 0.], [0., 0.], [0., 0.],
-                             [0., 0.], [0., 0.], [0., 0.], [0., 0.], [0., 0.],
-                             [0., 0.], [0., 0.], [0., 0.]]),
-                           ])
-    def test_common_start_returns(self, before, after, mean_by_date, demeaned,
+    @parameterized.expand([
+        (
+            2, 3, False, False,
+            [[4.93048307, 8.68843922], [6.60404312, 12.22369139],
+             [8.92068367, 17.1794088], [12.1275523, 24.12861778],
+             [16.5694159, 33.8740100], [22.7273233, 47.53995233]],
+        ),
+        (
+            3, 2, False, True,
+            [[0.0, 5.63219176], [0.0, 7.96515233],
+             [0.0, 11.2420646], [0.0, 15.8458720],
+             [0.0, 22.3134160], [0.0, 31.3970961]],
+        ),
+        (
+            3, 5, True, False,
+            [[3.7228318, 2.6210478], [4.9304831, 3.6296796], [6.6040431, 5.0193734],  # noqa
+             [8.9206837, 6.9404046], [12.127552, 9.6023405], [16.569416, 13.297652],  # noqa
+             [22.727323, 18.434747], [31.272682, 25.584180], [34.358565, 25.497254]],  # noqa
+        ),
+        (
+            1, 4, True, True,
+            [[0., 0.], [0., 0.], [0., 0.],
+             [0., 0.], [0., 0.], [0., 0.]],
+        ),
+        (
+            6, 6, False, False,
+            [[2.02679565, 2.38468223], [2.38769454, 3.22602748],
+             [2.85413029, 4.36044469], [3.72283181, 6.16462715],
+             [4.93048307, 8.68843922], [6.60404312, 12.2236914],
+             [8.92068367, 17.1794088], [12.1275523, 24.1286178],
+             [16.5694159, 33.8740100], [22.7273233, 47.5399523],
+             [31.2726821, 66.7013483], [34.3585654, 70.1828776],
+             [37.9964585, 74.3294620]],
+        ),
+        (
+            6, 6, False, True,
+            [[0.0, 2.20770299], [0.0, 2.95942924], [0.0, 3.97022414],
+             [0.0, 5.63219176], [0.0, 7.96515233], [0.0, 11.2420646],
+             [0.0, 15.8458720], [0.0, 22.3134160], [0.0, 31.3970962],
+             [0.0, 44.1512888], [0.0, 62.0533954], [0.0, 65.8668371],
+             [0.0, 70.4306483]],
+        ),
+        (
+            6, 6, True, False,
+            [[2.0267957, 0.9562173], [2.3876945, 1.3511898], [2.8541303, 1.8856194],  # noqa
+             [3.7228318, 2.6210478], [4.9304831, 3.6296796], [6.6040431, 5.0193734],  # noqa
+             [8.9206837, 6.9404046], [12.127552, 9.6023405], [16.569416, 13.297652],  # noqa
+             [22.727323, 18.434747], [31.272682, 25.584180], [34.358565, 25.497254],  # noqa
+             [37.996459, 25.198051]],
```

Comment:
It's unfortunate the diff here is so hard to read. This test (`test_common_returns`) and the one below (`test_average_cumulative_return_by_quantile`) had pretty significant changes, but these are expected by @jmccorriston and I according the changes [in this function](https://github.com/quantopian/alphalens/pull/361/files#diff-39e40ed256ddc7619de160850cb6f6a2R642) which uses returns now instead of pricing, and flips the interpretation of the `cumulative` parameter.